In [58]:
# Importar librerías para manipulación y análisis de datos
# -----------------------------------------------------------------------
import pandas as pd
from pandas import DataFrame
import numpy as np
pd.set_option('display.max_columns', None)  # Establece una opción d

In [59]:
df = pd.read_csv("ficheros/recursos-humanos.csv", index_col=0)

In [60]:
df.columns

Index(['Employee_Name', 'EmpID', 'MarriedID', 'MaritalStatusID', 'GenderID',
       'EmpStatusID', 'DeptID', 'PerfScoreID', 'FromDiversityJobFairID',
       'Salary', 'Termd', 'PositionID', 'Position', 'State', 'Zip', 'DOB',
       'Sex', 'MaritalDesc', 'CitizenDesc', 'HispanicLatino', 'RaceDesc',
       'DateofHire', 'DateofTermination', 'TermReason', 'EmploymentStatus',
       'Department', 'ManagerName', 'ManagerID', 'RecruitmentSource',
       'PerformanceScore', 'EngagementSurvey', 'EmpSatisfaction',
       'SpecialProjectsCount', 'LastPerformanceReview_Date', 'DaysLateLast30',
       'Absences', 'date', 'Remuneration', 'Age', 'Date_of_Birth'],
      dtype='object')

In [61]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 410 entries, 0 to 409
Data columns (total 40 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   Employee_Name               164 non-null    object 
 1   EmpID                       311 non-null    float64
 2   MarriedID                   257 non-null    float64
 3   MaritalStatusID             311 non-null    float64
 4   GenderID                    311 non-null    float64
 5   EmpStatusID                 311 non-null    float64
 6   DeptID                      311 non-null    float64
 7   PerfScoreID                 282 non-null    float64
 8   FromDiversityJobFairID      156 non-null    float64
 9   Salary                      311 non-null    object 
 10  Termd                       311 non-null    float64
 11  PositionID                  311 non-null    float64
 12  Position                    311 non-null    object 
 13  State                       311 non

In [68]:
df["DOB"].value_counts()

09/09/65    2
06/14/87    2
09/22/76    2
07/07/84    2
05/12/80    1
           ..
10/01/90    1
12/02/83    1
09/15/85    1
07/06/68    1
08/17/78    1
Name: DOB, Length: 307, dtype: int64

In [ ]:
fecha_datetime = datetime.strptime(fecha_string, '%m/%d/%Y')

In [69]:
df["DOB"].isnull().sum()

99

In [ ]:
def fecha(valor):
    try:
        return date(valor)
    except:
        return np.nan

In [ ]:
def fecha(valor):
    

1. 'EngagementSurvey' 
* puntuación de la encuesta de compromiso del empleado - Valores contínuos del 1.12 - 5.0 
* 99 nulos
* Float - ok

2. 'EmpSatisfaction' - Satisfacción del empleado. 
* Guess - Rango valores: 1 poco satisfecho al 5 muy satisfecho
* 311 valores
* Float - ok

3. 'SpecialProjectsCount': Número de proyectos especiales en los que ha participado el empleado
* Rango: del 0 al 8
* Nulos: 212
* Float - ok
** Analizar si los que siguen en la empresa participan todos en proyectos especiales

4. 'LastPerformanceReview_Date': Fecha de la última revisión de rendimiento del empleado
* Object - cambiar a date
* Nulos: 167
** volve a analizar una vez eliminados los duplicados

5. 'DaysLateLast30': Número de días de retraso en el último mes
* Float - ok
* Nulos: 219 -- borrar quizás
** Aclarar: retraso físico o de entrega de productos

6. 'Absences': Nro. de ausencias del empleado
* Tenemos 311 valores
* Float - ok
* Rango: 1 - 20

7. 'date': No hay definición de la columna
* Son 5 fechas distintas
* Nulos - 311 (tenemos valor para los 99 - quizás son de los que no tenemos employee ID?)
* Object - si la mantenemos, cambiar a fecha
* Borrar

8. 'Remuneration'
* Valores para 311. 99 nulos
* Object - cambiarlo a float
* Duplicado de "Salary" - borrarla

9. 'Age'
* Completar con la fecha de nacimiento (DoB)
* Muchos nulos (408), sólo 2 valores erróneos (-5, 150)

10. 'Date_of_Birth' - Eliminar. Sólo 23 valores y son inválidos. Info en DoB

In [65]:
df[["Remuneration","Salary"]].head(100)

Remuneration      Salary
0     62,506.00   62,506.00
1    104,437.00  104,437.00
2     64,955.00   64,955.00
3     64,991.00   64,991.00
4     50,825.00   50,825.00
..          ...         ...
95    70,187.00   70,187.00
96   178,000.00  178,000.00
97    99,351.00   99,351.00
98    67,251.00   67,251.00
99    65,707.00   65,707.00

[100 rows x 2 columns]

In [62]:
df["Age"].value_counts()

-5.0      1
 150.0    1
Name: Age, dtype: int64

In [63]:
because i didn´t know what to say

SyntaxError: invalid syntax (925035333.py, line 1)

In [ ]:
df['Remuneration'].value_counts()

63,025.00    2
57,815.00    2
61,242.00    2
66,738.00    1
68,829.00    1
            ..
51,337.00    1
53,171.00    1
52,249.00    1
65,707.00    1
45,046.00    1
Name: Remuneration, Length: 308, dtype: int64

In [ ]:
df['Absences'].max()

20.0

In [ ]:
df.info()


<class 'pandas.core.frame.DataFrame'>
Int64Index: 410 entries, 0 to 409
Data columns (total 40 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   Employee_Name               164 non-null    object 
 1   EmpID                       311 non-null    float64
 2   MarriedID                   257 non-null    float64
 3   MaritalStatusID             311 non-null    float64
 4   GenderID                    311 non-null    float64
 5   EmpStatusID                 311 non-null    float64
 6   DeptID                      311 non-null    float64
 7   PerfScoreID                 282 non-null    float64
 8   FromDiversityJobFairID      156 non-null    float64
 9   Salary                      311 non-null    object 
 10  Termd                       311 non-null    float64
 11  PositionID                  311 non-null    float64
 12  Position                    311 non-null    object 
 13  State                       311 non

In [ ]:
df['Age'].isnull().sum()

408

In [ ]:
df["date"].value_counts()

2022-01-03    23
2022-04-01    22
2022-05-01    19
2022-01-02    19
2022-01-01    16
Name: date, dtype: int64

In [ ]:
df["EngagementSurvey"].max()

5.0